frontend-public/src/components/dashboard/mine/variances/Variances.js

```diff
@@ -49,6 +68,80 @@ export class Variances extends Component {
     this.props.fetchVarianceStatusOptions();
   }
 
+  handlAddDocuments = (files, varianceGuid) => {
+    Promise.all(
+      Object.entries(files).map(([document_manager_guid, document_name]) =>
+        this.props.addDocumentToVariance(this.props.mine.mine_guid, varianceGuid, {
+          document_manager_guid,
+          document_name,
+        })
+      )
+    );
+  };
+
+  handleAddVariances = (files) => (values) => {
+    const received_date = values.received_date
+      ? values.received_date
+      : moment().format("YYYY-MM-DD");
+    const newValues = { received_date, ...values };
+    return this.props
+      .createVariance(this.props.mine.mine_guid, this.props.mine.mine_name, newValues)
+      .then(async ({ data: { variance_guid } }) => {
+        await this.handlAddDocuments(files, variance_guid);
```

Comment:
This is where the issue of `handlAddDocuments` not returning a Promise would come up. It'll be a race condition against the following actions

Reply:
Returned it, everything is working as expected now!